Shapes/rml_rules_shape.ttl

```diff
@@ -0,0 +1,1683 @@
+        Exactly one rr:template, one rr:column, one rr:constant or one 
+        rml:reference is required.
+        """ ;
+            ns1:minCount 1 ;
+            ns1:name "rr:template/rr:column/rr:constant/rml:reference" ;
+            ns1:path [ ns1:alternativePath ( <http://www.w3.org/ns/r2rml#template> <http://www.w3.org/ns/r2rml#column> <http://www.w3.org/ns/r2rml#constant> <http://semweb.mmlab.be/ns/rml#reference> ) ] ],
+        [ ns1:datatype xsd:string ;
+            ns1:description """
+        A template (format string) to specify how to generate a value for a 
+        subject, predicate, or object, using one or more columns from a logical
+        table row or iterator.
+        """ ;
+            ns1:message """
+        rr:template must be a string.
+        """ ;
+            ns1:name "rr:template" ;
+            ns1:nodeKind ns1:Literal ;
+            ns1:path <http://www.w3.org/ns/r2rml#template> ],
+        [ ns1:datatype xsd:string ;
+            ns1:description """
+        Name of a column in the logical table. When generating RDF triples from
+        a logical table row, value from the specified column is used as the 
+        subject, predicate, or object (based upon the specific domain).
+        """ ;
+            ns1:message """
+        rr:column must be a string.
+        """ ;
+            ns1:name "rr:column" ;
+            ns1:nodeKind ns1:Literal ;
+            ns1:path <http://www.w3.org/ns/r2rml#column> ],
+        [ ns1:description """
+        A property for indicating whether a term map is a constant-valued term 
+        map.
+        """ ;
+            ns1:message """
+        rr:constant must be an IRI.
+        """ ;
+            ns1:name "rr:constant" ;
+            ns1:nodeKind ns1:IRI ;
+            ns1:path <http://www.w3.org/ns/r2rml#constant> ],
+        [ ns1:datatype xsd:string ;
+            ns1:description """
+        A reference rml:reference is used to refer to a column in case of 
+        databases, a record in case of CSV or TSV data source, an element in 
+        case of XML data source, an object in case of a JSON data source, etc.
+
+        A reference must be a valid identifier, considering the reference 
+        formulation (rml:referenceFormulation) specified. The reference can be
+        an absolute path, or a path relative to the iterator specified at the 
+        logical source. 
+        """ ;
+            ns1:message """
+        rml:reference must be a string.
+        """ ;
+            ns1:name "rml:reference" ;
+            ns1:nodeKind ns1:Literal ;
+            ns1:path <http://semweb.mmlab.be/ns/rml#reference> ] .
+
+<http://schema.org/RRobjectShape> a ns1:NodeShape ;
+    ns1:closed true ;
+    ns1:description """
+    An IRI reference for use as object for all the RDF triples generated from 
+    a logical table row or iterator.
+    """ ;
+    ns1:ignoredProperties ( rdf:type ) ;
+    ns1:message """
+    rr:object shortcut for rr:objectMap must be an IRI.
```

Comment:
`rr:object` can be used with any node kind I think?

Reply:
Right! That's one of these shortcuts... Probably a copy-paste error.